activestorage/lib/active_storage/attached/changes/create_many.rb

```diff
@@ -29,7 +29,9 @@ def save
 
     private
       def subchanges
-        @subchanges ||= attachables.collect { |attachable| build_subchange_from(attachable) }
+        @subchanges ||= attachables.collect { |attachable| build_subchange_from(attachable) }.tap do |subchanges|
+          @pending_upload += subchanges.reject { |subchange| subchange.attachable.is_a?(ActiveStorage::Blob) }
```

Comment:
It's a bit implicit that invoking `#subchanges` will accumulate the pending uploads. Maybe we can do this explicitly in the constructor:

```ruby

def initialize
...
  @pending_upload += subchanges_without_blobs
...
end

private
  ...
  def subchanges_without_blobs
    subchanges.reject { |subchange| subchange.attachable.is_a?(ActiveStorage::Blob) }
  end
```